ReactiveCocoa/Swift/Property.swift

```diff
@@ -118,6 +118,37 @@ public final class MutableProperty<Value>: MutablePropertyType {
 		observer.sendNext(initialValue)
 	}
 
+	/// Atomically replaces the contents of the variable.
+	///
+	/// Returns the old value.
+	public func swap(newValue: Value) -> Value {
+		return modify { _ in newValue }
+	}
+
+	/// Atomically modifies the variable.
+	///
+	/// Returns the old value.
+	public func modify(@noescape action: (Value) throws -> Value) rethrows -> Value {
+		lock.lock()
+		defer { lock.unlock() }
+
+		let oldValue = _value
+		_value = try action(_value)
+		observer.sendNext(_value)
+		return oldValue
+	}
+
+	/// Atomically performs an arbitrary action using the current value of the
+	/// variable.
+	///
+	/// Returns the result of the action.
+	public func withValue<Result>(@noescape action: (Value) throws -> Result) rethrows -> Result {
```

Comment:
Using `Result` as a generic type parameter here may be confusing, since `Result` already exists. Maybe `Output`?